lib/resource-providers/vpc.ts

```diff
@@ -29,9 +35,27 @@ export class VpcProvider implements ResourceProvider<ec2.IVpc> {
             // It will automatically divide the provided VPC CIDR range, and create public and private subnets per Availability Zone.
             // Network routing for the public subnets will be configured to allow outbound access directly via an Internet Gateway.
             // Network routing for the private subnets will be configured to allow outbound access via a set of resilient NAT Gateways (one per AZ).
+            // Creates Secondary CIDR and Secondary subnets if passed.
             vpc = new ec2.Vpc(context.scope, id + "-vpc");
+            const secondarySubnets: Array<PrivateSubnet> = [];
+            if (this.secondaryCidr) {
+                new ec2.CfnVPCCidrBlock(context.scope, id + "-secondaryCidr", {
+                    vpcId: vpc.vpcId,
+                    cidrBlock: this.secondaryCidr});
+                if (this.secondarySubnetCidrs) {
+                    for(let az in vpc.availabilityZones) {
+                        secondarySubnets[az] = new ec2.PrivateSubnet(context.scope, id + "private-subnet-" + vpc.availabilityZones[az], {
```

Comment:
I am unclear how do you plan to pass these subnet IDs to the vpc-cni add-on. Let's discuss it. I am sure customers will want just that.